pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -144,24 +142,26 @@ func (npc *NetworkPolicyController) Run(healthChan chan<- *healthcheck.Controlle
 	glog.Info("Starting network policy controller")
 	npc.healthChan = healthChan
 
-	// loop forever till notified to stop on stopCh
-	for {
-		select {
-		case <-stopCh:
-			glog.Info("Shutting down network policies controller")
-			return
-		default:
+	// Full syncs of the network policy controller take a lot of time and can only be processed one at a time,
+	// therefore, we start it in it's own goroutine and request a sync through a single item channel
+	glog.Info("Starting network policy controller full sync goroutine")
+	go func(fullSyncRequest <-chan struct{}, stopCh <-chan struct{}) {
+		for {
+			select {
+			case <-stopCh:
+				glog.Info("Shutting down network policies full sync goroutine")
+				return
```

Comment:
we might as well add this goroutine to WaitGroup

Reply:
I created an additional WaitGroup here and then wait on it before the WaitGroup passed to Run sends done. Is that ok? Or did you want me to piggyback off the initial WaitGroup that was passed to Run()?

Reply:
@aauren I meant to the existing waitgroup that was passed to Run().  So that whole process will exit gracefully (by waiting here https://github.com/cloudnativelabs/kube-router/blob/v1.0.0-rc3/pkg/cmd/kube-router.go#L190) when all the go routines exits.

Reply:
Updated to use the original waitgroup